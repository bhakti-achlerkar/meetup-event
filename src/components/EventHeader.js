import React from 'react';
import PropTypes from 'prop-types';
import EventVenue from './EventVenue';
import { formatDate, formatDuration } from '../utils';

const EventHeader = ({ duration = '5400000',
  eventName='',
  eventDate,
  groupName='',
  venue={} }) => {
  return (
    <header className="event-header">
      <h1>{eventName}</h1>
      <h3 className="event-group">By {groupName}</h3>
      <div className="event-time-venue">
        <div className="event-time-details">
          <i className="fas fa-clock fa-lg"></i>
          <div>
            <div>{formatDate(eventDate)}</div>
            <div>{formatDuration(eventDate, duration)}</div>
          </div>
        </div>
        <EventVenue venue={venue}></EventVenue>
      </div>
    </header>
  );
};

EventHeader.propTypes = {
  duration: PropTypes.string,
  eventName: PropTypes.string.isRequired,
  groupName: PropTypes.string.isRequired,
  eventDate: PropTypes.number,
  venue: PropTypes.object
};

export default EventHeader;
