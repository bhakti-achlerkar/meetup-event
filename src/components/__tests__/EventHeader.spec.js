import React from 'react';
import EventHeader from '../EventHeader';
import renderer from 'react-test-renderer';


test('EventHeader rendered with event details', () => {
  const component = renderer.create(
    <EventHeader
      eventName="EVENT_NAME"
      groupName="GROUP_NAME"
      venue={{ name: "VENUE_NAME" }}></EventHeader>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
