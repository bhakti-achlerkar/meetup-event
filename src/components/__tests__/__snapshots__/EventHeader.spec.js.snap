// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHeader rendered with event details 1`] = `
<header
  className="event-header"
>
  <h1>
    EVENT_NAME
  </h1>
  <h3
    className="event-group"
  >
    By 
    GROUP_NAME
  </h3>
  <div
    className="event-time-venue"
  >
    <div
      className="event-time-details"
    >
      <i
        className="fas fa-clock fa-lg"
      />
      <div>
        <div>
          Invalid Date
        </div>
        <div>
          NaN.NaNAM to NaN.NaNAM
        </div>
      </div>
    </div>
    <div
      className="event-host"
    >
      <span>
        <i
          className="fas fa-map-marker fa-lg"
        />
      </span>
      <div>
        <div>
           
          VENUE_NAME
        </div>
        <div
          className="event-address"
        >
          <span />
          <span />
        </div>
        <div
          className="event-address"
        >
          , 
           - 
        </div>
      </div>
    </div>
  </div>
</header>
`;
